client/src/main/java/io/pravega/client/stream/impl/ControllerResolverFactory.java

```diff
@@ -65,7 +69,7 @@ public NameResolver newNameResolver(URI targetUri, Attributes params) {
         final String authority = targetUri.getAuthority();
         final List<InetSocketAddress> addresses = Splitter.on(',').splitToList(authority).stream().map(host -> {
             final String[] strings = host.split(":");
-            return new InetSocketAddress(strings[0], Integer.valueOf(strings[1]));
+            return InetSocketAddress.createUnresolved(strings[0], Integer.valueOf(strings[1]));
```

Comment:
If ControllerNameResolves takes a list of URI instead of InetSocketAddress, we can ensure we create the InetSocketAddress inside getControllers method.